src/client/components/stores/TypesStore.ts

```diff
@@ -1,24 +1,28 @@
 import { ReduceStore } from "flux/utils";
+import * as Immutable from 'immutable';
 import TypeDispatcher from "../../dispatcher/TypeDispatcher";
-import { TypeActions } from "../../constants/TypeActions";
+import { TypeActionTypes } from "../../constants/TypeActionTypes";
+import {Type} from "../../models/Type";
 
-// export default class TypesStore extends ReduceStore {
-//     constructor() {
-//         super(TypeDispatcher);
-//         this.state = {};
-//     }
-//
-//     reduce(state, action) {
-//         switch (action.eventName) {
-//             case TypeActions.ADD_TYPE:
-//                 return state;
-//
-//             default:
-//                 return state;
-//         }
-//     }
-//
-//
-// }
 
-export default {};
\ No newline at end of file
+class TypesStore extends ReduceStore<Immutable.List<Type>, any> {
+    constructor() {
+	    super(TypeDispatcher);
+    }
+
+	getInitialState(): Immutable.List<Type> {
```

Comment:
pls, fix indentation
Also can we use just typescript features instead Immutable